streaming/base/shared.py

```diff
@@ -27,37 +33,45 @@ class SharedBarrier:
     Args:
         filelock_path (str): Path to lock file on local filesystem.
         shm_path (str): Shared memory object name in /dev/shm.
+        to_create (bool): Create a new shared memory block or attaches to an existing shared memory block.
     """
 
-    def __init__(self, filelock_path: str, shm_path: str) -> None:
+    def __init__(self, filelock_path: str, shm_path: str, to_create: bool) -> None:
         self.filelock_path = filelock_path
         self.shm_path = shm_path
 
-        # Create filelock.
-        dirname = os.path.dirname(filelock_path)
-        os.makedirs(dirname, exist_ok=True)
-        self.lock = FileLock(filelock_path)
-
-        # Create three int32 fields in shared memory: num_enter, num_exit, flag.
-        size = 3 * np.int32(0).nbytes
-        try:
-            self._shm = SharedMemory(shm_path, True, size)
-        except FileExistsError:
-            sleep(TICK)
-            self._shm = SharedMemory(shm_path, False, size)
+        start_time = time()
+        while True:
+            try:
+                # Create three int32 fields in shared memory: num_enter, num_exit, flag.
+                size = 3 * np.int32(0).nbytes
+                self._shm = SharedMemory(shm_path, to_create, size)
+
+                # Create filelock.
+                self.dirname = os.path.dirname(filelock_path)
+                os.makedirs(self.dirname, exist_ok=True)
+                self.lock = FileLock(filelock_path)
+                break
+            except FileNotFoundError:
+                sleep(TICK)
+                dt = time() - start_time
```

Comment:
"dt" -> "elapsed"

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/107/commits/ce88df6c54e914e82797c6ab504f4a01012a742a.